pandas/tools/merge.py

```diff
@@ -769,9 +770,16 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
             if keys is None:
                 keys = sorted(objs)
             objs = [objs[k] for k in keys]
+            obj_count = len(objs)
+        else:
+            count_gen, objs = itertools.tee(objs)
+            obj_count = len(list(count_gen))
+
+        if obj_count == 0:
```

Comment:
Well, it does look a little hackish. I had issues with `objs` being a generator. `len(objs)` would consume the generator, that's why I used the hack with `itertools.tee`.